### PR TITLE
Standardized http api

### LINKED ISSUE #670: [Feature] Standardized http api
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

http api

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by yukkit**
# Which issue does this PR close?

Closes #670 .

# Rationale for this change

# What changes are included in this PR?

- Modify http api
- Adapt to the integration test framework
- Add integration test step in github ci

# Are there any user-facing changes?

Refer to the issue #670  description

### CODE REVIEW

File: main/src/http/header.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    pub fn get_accept(&self) -> &str {
+        self.accept.as_deref().unwrap_or(APPLICATION_CSV)
+    }
+
+    pub fn try_get_basic_auth(&self) -> Result<UserInfo, HttpError> {
+        let auth = &self.authorization;
+
+        let err = Err(HttpError::ParseAuth {
```
Comment (zipper-meng) [inline main/src/http/header.rs:41]: How about using this?
```rust
let get_err = || Err(HttpError::ParseAuth {
    reason: auth.to_string(),
});
```

Comment (yukkit) [reply main/src/http/header.rs:41]: The same exception is returned in several places, so initialization is performed at the beginning

Comment (yukkit) [reply main/src/http/header.rs:41]: I agree

Comment (yukkit) [reply main/src/http/header.rs:41]: Can I hear your thoughts

**Review by zipper-meng — APPROVED** [review]
LGTM

**Comment by zipper-meng** [comment]
```sh
echo user1:password1|base64
# dXNlcjE6cGFzc3dvcmQxCg==
```

```sh
curl -i -XPOST -H "AUTHORIZATION: Basic dXNlcjE6cGFzc3dvcmQxCg==" 'http://127.0.0.1:31007/api/v1/write?db=dba' -d 'ma,ta=a1,tb=b1 fa=1,fb=1'
# HTTP/1.1 200 OK
# content-length: 0
```

```sh
curl -i -XPOST -H "AUTHORIZATION: Basic dXNlcjE6cGFzc3dvcmQxCg==" 'http://127.0.0.1:31007/api/v1/sql?db=dba' -d 'select * from ma'
# HTTP/1.1 200 OK
# content-type: application/csv
# content-length: 61
```

**Pull request merged**